packages/extension/src/shared/token/storage.ts

```diff
@@ -51,3 +60,50 @@ export async function removeToken(token: BaseToken) {
   await assertSchema(baseTokenSchema, token)
   return tokenStore.remove((t) => equalToken(t, token))
 }
+
+export async function updateTokenList() {
+  const now = new Date()
+  const nonAlphTokenList = await tokenListStore.get((t) => t.id !== ALPH_TOKEN_ID)
+
+  if (nonAlphTokenList[0]?.updatedAt) {
+    const updatedSinceInMinutes = (now.valueOf() - nonAlphTokenList[0].updatedAt) / 1000 / 60
+    // Update token list at most every hour
+    if (updatedSinceInMinutes < 59) {
```

Comment:
Question:

1. We have already set the alarm to fire every hour, why is this check necessary here?
2. If this condition is true, does this imply that the time interval for this update will be two hours?

Reply:
> We have already set the alarm to fire every hour, why is this check necessary here?
Since we try to update in a few places:
1. Alarm starts and fires every one hour
2. When wallet is opened
3. When user clicks the "update now" button
So even thought alarms is fired every one hour, we actually triggers this more often that that, hence the check.

> If this condition is true, does this imply that the time interval for this update will be two hours?
The worst case scenario yes, but since once user open the wallet there will be a update, so probably user wont notice that. WDTY?